lib/ruby_lsp/tapioca/server_addon.rb

```diff
@@ -23,6 +25,34 @@ def dsl(params)
         load("tapioca/cli.rb") # Reload the CLI to reset thor defaults between requests
         ::Tapioca::Cli.start(["dsl", "--lsp_addon", "--workers=1"] + params[:constants])
       end
+
+      def gem(params)
+        snapshot_specs = parse_lockfile(params[:snapshot_lockfile])
+        current_specs = parse_lockfile(params[:current_lockfile])
+
+        removed_gems = snapshot_specs.keys - current_specs.keys
+        changed_gems = current_specs.select { |name, version| snapshot_specs[name] != version }.keys
+
+        return $stdout.puts("No gem changes detected") if removed_gems.empty? && changed_gems.empty?
```

Comment:
Let's put the `return` on its own line, since returning the result of `puts` isn't meaningful.